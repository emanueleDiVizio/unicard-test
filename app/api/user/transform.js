export const transformGetUsers = ({ok, data, problem}) => {
  const newData = ok
    ? data.data.map(({avatar, first_name, last_name, email}) => ({
        name: first_name,
        image: avatar,
        surname: last_name,
        email,
      }))
    : {};

  return {ok, data: newData, problem};
};
