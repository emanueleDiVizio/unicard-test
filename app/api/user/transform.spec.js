import {transformGetUsers} from './transform';

const MOCK_DATA = [
  {
    id: 1,
    email: 'george.bluth@reqres.in',
    first_name: 'George',
    last_name: 'Bluth',
    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg',
  },
  {
    id: 2,
    email: 'janet.weaver@reqres.in',
    first_name: 'Janet',
    last_name: 'Weaver',
    avatar:
      'https://s3.amazonaws.com/uifaces/faces/twitter/josephstein/128.jpg',
  },
  {
    id: 3,
    email: 'emma.wong@reqres.in',
    first_name: 'Emma',
    last_name: 'Wong',
    avatar:
      'https://s3.amazonaws.com/uifaces/faces/twitter/olegpogodaev/128.jpg',
  },
  {
    id: 4,
    email: 'eve.holt@reqres.in',
    first_name: 'Eve',
    last_name: 'Holt',
    avatar:
      'https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg',
  },
  {
    id: 5,
    email: 'charles.morris@reqres.in',
    first_name: 'Charles',
    last_name: 'Morris',
    avatar:
      'https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg',
  },
  {
    id: 6,
    email: 'tracey.ramos@reqres.in',
    first_name: 'Tracey',
    last_name: 'Ramos',
    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg',
  },
];

describe('api/user/transform', () => {
  describe('transformGetUsers', () => {
    it('transforms data correctly when response is a success', () => {
      const mockResponse = {
        ok: true,
        data: {data: MOCK_DATA},
      };
      const transformed = transformGetUsers(mockResponse);

      expect(transformed).toMatchSnapshot();
    });
  });

  it('transforms data correctly when response has a problem', () => {
    const mockProblem = 'test';
    const mockResponse = {
      ok: false,
      problem: mockProblem,
    };
    const transformed = transformGetUsers(mockResponse);

    expect(transformed).toMatchSnapshot();
  });
});
