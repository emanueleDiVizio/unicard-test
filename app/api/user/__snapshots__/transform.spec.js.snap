// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/user/transform transformGetUsers transforms data correctly when response is a success 1`] = `
Object {
  "data": Array [
    Object {
      "email": "george.bluth@reqres.in",
      "image": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg",
      "name": "George",
      "surname": "Bluth",
    },
    Object {
      "email": "janet.weaver@reqres.in",
      "image": "https://s3.amazonaws.com/uifaces/faces/twitter/josephstein/128.jpg",
      "name": "Janet",
      "surname": "Weaver",
    },
    Object {
      "email": "emma.wong@reqres.in",
      "image": "https://s3.amazonaws.com/uifaces/faces/twitter/olegpogodaev/128.jpg",
      "name": "Emma",
      "surname": "Wong",
    },
    Object {
      "email": "eve.holt@reqres.in",
      "image": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
      "name": "Eve",
      "surname": "Holt",
    },
    Object {
      "email": "charles.morris@reqres.in",
      "image": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
      "name": "Charles",
      "surname": "Morris",
    },
    Object {
      "email": "tracey.ramos@reqres.in",
      "image": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
      "name": "Tracey",
      "surname": "Ramos",
    },
  ],
  "ok": true,
  "problem": undefined,
}
`;

exports[`api/user/transform transforms data correctly when response has a problem 1`] = `
Object {
  "data": Object {},
  "ok": false,
  "problem": "test",
}
`;
